src/impl/realm_coordinator.hpp

```diff
@@ -145,7 +146,9 @@ class RealmCoordinator : public std::enable_shared_from_this<RealmCoordinator> {
     std::unique_ptr<_impl::ExternalCommitHelper> m_notifier;
     std::function<void(VersionID, VersionID)> m_transaction_callback;
 
+#ifdef REALM_ENABLE_SYNC
     std::shared_ptr<SyncSession> m_sync_session;
```

Comment:
Because the Java binding distribute the core library without any sync code. So we are stripping all references to sync code to reduce the size.